frontend/src/app/general/nav-bar/nav-bar.service.ts

```diff
@@ -5,20 +5,65 @@
 
 import { Injectable } from '@angular/core';
 import { MatSidenav } from '@angular/material/sidenav';
+import { BehaviorSubject, map, Observable, tap } from 'rxjs';
+import {
+  BuiltInLinkItem,
+  NavbarConfigurationOutput,
+  NavbarService as OpenAPINavbarService,
+} from 'src/app/openapi';
 import { UserRole } from 'src/app/services/user/user.service';
 import { environment } from 'src/environments/environment';
 
 @Injectable({
   providedIn: 'root',
 })
 export class NavBarService {
-  sidenav?: MatSidenav;
+  constructor(private navbarService: OpenAPINavbarService) {
+    this.loadNavbarConfig().subscribe();
+  }
+
+  loadNavbarConfig(): Observable<NavbarConfigurationOutput> {
+    return this.navbarService
+      .getNavbar()
+      .pipe(tap((navConf) => this._navbarConfig.next(navConf)));
+  }
+
+  private _navbarConfig = new BehaviorSubject<
+    NavbarConfigurationOutput | undefined
+  >(undefined);
 
+  readonly navbarItems$ = this._navbarConfig.pipe(
+    map(
+      (navbarConfig): NavBarItem[] =>
+        navbarConfig?.external_links.map((link) => {
+          if (link.service) {
+            return {
+              name: link.name,
+              href: this._linkMap[link.service],
+              target: '_blank',
+              icon: 'open_in_new',
+              requiredRole: link.role,
+            };
+          } else {
+            return {
+              name: link.name,
+              href: link.href,
+              target: '_blank',
+              icon: 'open_in_new',
+              requiredRole: link.role,
+            };
+          }
```

Comment:
```suggestion
          return {
            name: link.name,
            href: link.service ? this._linkMap[link.service] : link.href,
            target: '_blank',
            icon: 'open_in_new',
            requiredRole: link.role,
          };
```

That's a lot simpler.